compiler/rustc_type_ir/src/new.rs

```diff
@@ -0,0 +1,19 @@
+use crate::{BoundVar, ConstKind, DebruijnIndex, Interner, RegionKind, TyKind};
+
+pub trait Ty<I: Interner<Ty = Self>> {
```

Comment:
I don't believe it's feasible to share code b/w this and `rustc_middle`, since I expect we don't want to import this `Ty` trait into hundreds and hundreds of modules in the compiler, and also this would conflict with the naming of `Ty` in `rustc_middle`.

This naming conflict is actually kind of intentional, since we can use `Ty::new_x(...)` in uplifted solver code without really needing to change much.

Reply:
time to figure out how to create custom preludes 🙃 

The duplication seems fine, though it would be neat if we could generate most of the boilerplate